src/components/va-pagination/va-pagination.tsx

```diff
@@ -0,0 +1,306 @@
+    if (opts.edges > 3) {
+      opts.edges = 3;
+    }
+    // edges * 2 + 2 ellipsis + 1 visible active link in the middle
+    let minVisible = opts.edges * 2 + 2 + 1;
+    if (minVisible > opts.total) {
+      minVisible = opts.total;
+    }
+    if (minVisible > opts.visible) {
+      opts.visible = minVisible;
+    }
+    return opts;
+  }
+
+  /**
+   * Set current page state & emit pageSelect event
+   * @param page - user selected page
+   */
+  private onPageSelected = (page: number) => {
+    this.currentPage = page;
+    this.pageSelect.emit(page);
+  };
+
+  /**
+   * Key down event handler - this allows the user to hold down the key to
+   * quickly navigate through the pages
+   * @param key - event.key
+   * @param page - page associated with the interaction
+   */
+  private handleKeyDown = (key: string, page: number) => {
+    const activate = ['Enter', ' ', 'Spacebar' /* IE11 */];
+    if (activate.includes(key)) {
+      this.onPageSelected(page);
+    }
+  };
+
+  /**
+   * Return common link properties
+   * @param page - page associated with the link
+   * @param current - "page" indicates that the link is for the current page
+   *   otherwise set to null so the attribute isn't included
+   * @param className - custom link class
+   */
+  private getLinkProps = ({
+    page,
+    current = '',
+    className = '',
+  }) => ({
+    href: '#',
+    class: className,
+    'data-page': page,
+    'aria-label': (this.linkAriaLabel || '').replace(/\{page\}/g, page),
+    'aria-current': current ? 'page' : null,
+    onClick: (event: MouseEvent) => {
+      event.preventDefault();
+      this.currentPage = page;
+      this.onPageSelected(page);
+    },
+    onKeyDown: (event: KeyboardEvent) => this.handleKeyDown(event.key, page)
+  });
+
+  /**
+   * Add previous link. Renders a span when there is no previous page to
+   * navigate to
+   * @param Options
+   * @returns {JSX} previous link
+   */
+  private addPrevLink = ({ current }) => {
+    const prevPage = current - 1 >= 1 ? current - 1 : null;
+    const content = (
+      <Fragment>
+        <span class="vads-u-visibility--screen-reader">Previous</span>
+        <span aria-hidden="true">{'\u2039\u00a0\u00a0'}
+          <span class="hide-narrow">Prev</span>
+        </span>
+      </Fragment>
+    );
+    if (prevPage === null) {
+      return <li aria-hidden="true"><span class="prev">{content}</span></li>;
+    }
+    const props = this.getLinkProps({
+      page: prevPage,
+      className: 'prev',
+    });
+    return <li><a {...props}>{content}</a></li>;
+  };
+
+    /**
+   * Add next link. Renders a span when there is no next page to navigate to
+   * @param Options
+   * @returns {JSX} next link
+   */
+  private addNextLink = ({ current, total }) => {
+    const nextPage = current + 1 <= total
+      ? current + 1
+      : null;
+    const content = (
+      <Fragment>
+        <span class="hide-narrow">Next</span>
+        {'\u00a0\u00a0\u203a'}
```

Comment:
What is this supposed to be?

Reply:
The `\u00a0` are non-breaking white spaces and the `\u203a` is a `›` character. This is added by the formation `nav-paginate-next` class name, but I ended up not using that class because it adds negative margins to keep the spaces and symbol in the view (narrow screens). The same is true for the previous link.